src/control/tools/EditSelection.cpp

```diff
@@ -35,7 +36,8 @@ EditSelection::EditSelection(UndoRedoHandler* undo, const PageRef& page, XojPage
     contstruct(undo, view, page);
 }
 
-EditSelection::EditSelection(UndoRedoHandler* undo, Selection* selection, XojPageView* view) {
+EditSelection::EditSelection(UndoRedoHandler* undo, Selection* selection, XojPageView* view):
+        SnapToGridInputHandler(view->getXournal()->getControl()->getSettings()) {
```

Comment:
Use composition over inheritance here, since this is a "has-an" not an "is-an" relationship.